controlplane/kubeadm/api/v1beta1/kubeadm_control_plane_types.go

```diff
@@ -228,6 +232,18 @@ type RemediationStrategy struct {
 	MinHealthyPeriod *metav1.Duration `json:"minHealthyPeriod,omitempty"`
 }
 
+// KubeadmControlPlaneNamingStrategy defines the naming strategy for control plane objects.
+type KubeadmControlPlaneNamingStrategy struct {
+	// Template defines the template to use for generating the name of the ControlPlane object.
+	// If not defined, it will fallback to `{{ .machine.name }}`.
+	// If the templated string exceeds 63 characters, it will be trimmed to 58 characters and will
+	// get concatenated with a random suffix of length 5.
+	// The templating mechanism provides the following arguments:
+	// * `.machine.name`: The name of the machine object.
+	// +optional
+	Template *string `json:"template,omitempty"`
+}
```

Comment:
```suggestion
// MachineNamingStrategy allows changing the naming pattern used when creating Machines.
// Note: InfraMachines & KubeadmConfigs will use the same name as the corresponding Machines.
type MachineNamingStrategy struct {
	// Template defines the template to use for generating the names of the Machine objects.
	// If not defined, it will fallback to `{{ .kubeadmControlPlane.name }}-{{ .random }}`.
	// If the templated string exceeds 63 characters, it will be trimmed to 58 characters and will
	// get concatenated with a random suffix of length 5.
	// The templating mechanism provides the following arguments:
	// * `.kubeadmControlPlane.name`: The name of the KubeadmControlPlane object.
	// * `.random`: A random alphanumeric string, without vowels, of length 5.
	// +optional
	Template *string `json:"template,omitempty"`
}
```